### PR TITLE
fix(src/es/Hackstore) Video empty

### PR DESCRIPTION
**Pull request by Dark25**
Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt [hunk 1]
```diff
@@ -195,22 +197,35 @@ class Hackstore : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
...
+    private fun extractUrlFromDonFunction(fullUrl: String): String {
+        val client = OkHttpClient()
+
+        val request = Request.Builder().url(fullUrl).build()
+        val response = client.newCall(request).execute()
+        val body = response.body.string()
+
+        val document = Jsoup.parse(body)
+
+        val scriptElement = document.select("script:containsData(function don())").firstOrNull()
```
Comment (Secozzi) [inline src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt:209]:
```suggestion
        val scriptElement = document.selectFirst("script:containsData(function don())")
```

File: src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt [hunk 2]
```diff
@@ -195,22 +197,35 @@ class Hackstore : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
     override fun episodeFromElement(element: Element): SEpisode { throw UnsupportedOperationException() }
 
     // ============================ Video Links =============================
+    private fun extractUrlFromDonFunction(fullUrl: String): String {
+        val client = OkHttpClient()
+
+        val request = Request.Builder().url(fullUrl).build()
+        val response = client.newCall(request).execute()
```
Comment (Secozzi) [inline src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt:204]:
```suggestion
        val response = client.newCall(GET(fullUrl, headers)).execute()
```
You should use the functions provided in `eu.kanade.tachiyomi.network` for making requests

File: src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt [hunk 3]
```diff
@@ -195,22 +197,35 @@ class Hackstore : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
     override fun episodeFromElement(element: Element): SEpisode { throw UnsupportedOperationException() }
 
     // ============================ Video Links =============================
+    private fun extractUrlFromDonFunction(fullUrl: String): String {
+        val client = OkHttpClient()
```
Comment (Secozzi) [inline src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt:201]: Is creating a new client really necessary? If it is, make it a class property instead so it won't create a new client each time the function is called.

File: src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt [hunk 4]
```diff
@@ -195,22 +197,35 @@ class Hackstore : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
...
-            val server = serverElement.text()
-            val url = linkElement.attr("href")
-
-            val isLatino = server.contains("latino")
-            val isSub = server.contains("subtitulado")
+        tabs.forEach { tab ->
+            val server = tab.select("a.playr").text()
+            val deco = tab.select("a.playr").attr("data-href")
+            val langs = tab.select("a.playr").attr("data-lang")
+            val fullUrl = baseUrl + deco
```
Comment (Secozzi) [inline src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt:225]:
```suggestion
            val fullUrl = tab.select("a.playr").attr("abs:data-href")
            val langs = tab.select("a.playr").attr("data-lang")
```

File: src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt [hunk 5]
```diff
@@ -195,22 +197,29 @@ class Hackstore : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
     override fun episodeFromElement(element: Element): SEpisode { throw UnsupportedOperationException() }
 
     // ============================ Video Links =============================
+    private fun extractUrlFromDonFunction(fullUrl: String): String {
+        val response = client.newCall(GET(fullUrl, headers)).execute()
+        val body = response.body.string()
+        val document = Jsoup.parse(body)
```
Comment (Secozzi) [inline src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt:203]:
```suggestion
        val document = response.asJsoup()
```

File: src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt (hunk 4 above)
Comment (Dark25) [reply src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt:219]: I tried this change, but it doesn't get the url

File: src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt (hunk 5 above)
Comment (Dark25) [reply src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt:202]: same here

**Review by jmir1 — APPROVED** [review]

**Pull request merged**